Fix server connection timeout

hickeyma opened the pull request:
This PR handles the following issues found with the `timeout` setting in the tools:

- The default server connection timeout is stated in documentation as 10s but in code it is set to 2s. This doc is now changed to 2s.
- It allows a negative timeout value  which is probably not a very good setting as the system will give it an indeterminate value. This means an unknown or indefinite timeout. This now uses the default value when set to negative.
- Finally, there seemed to some bugs the way the timeout value is retrieved and handled from config file and CLI. It doesn't seem to be converted to seconds as expected.

Partial #605 

[EDITED]

hickeyma commented:
/assign @yujuhong

hickeyma commented:
/assign @feiskyer

mikebrow commented:
F0511 21:19:23.550924   21939 server.go:152] unknown flag: --experimental-dockershim

mikebrow commented:
sorry @hickeyma I missed this the first time I skimmed the logs.. doing to many things at once

so there's more than one issue... the timeout value was one.. but unrelated to the travis fail.

mikebrow commented:
https://github.com/kubernetes/kubernetes/commit/53adde65ce000c4d90ee8f807e90658426733a52#diff-41db1a9454b65c65b4c44937886b5880

hickeyma commented:
@mikebrow Thanks for the review. Updated and ready for review again. Description also updated and logic changed from initial commit.

hickeyma commented:
> see https://github.com/kubernetes-sigs/cri-tools/blob/master/hack/run-critest.sh#L33 looks like k8s finally removed that flag..

Thanks for heads up @mikebrow. PR #607 now opened for that issue.

hickeyma commented:
CI now passing. PR is good for review again.

saschagrunert commented on cmd/crictl/main.go:
```diff
@@ -196,12 +203,12 @@ func main() {
 			Aliases: []string{"t"},
 			Value:   defaultTimeout,
-			Usage:   "Timeout of connecting to the server",
+			Usage:   "Timeout of connecting to the server (default: 2s)",
```
```suggestion
			Usage:   fmt.Sprintf("Timeout of connecting to the server (default: %ds)", defaultTimeout),
```

hickeyma replied:
Thanks for the suggestion. Seemed to miss this previously but it is not needed as CLI lib prints the default.

saschagrunert commented on cmd/crictl/main.go:
```diff
@@ -196,12 +203,12 @@ func main() {
 			Name:    "debug",
 			Aliases: []string{"D"},
-			Usage:   "Enable debug mode",
+			Usage:   "Enable debug mode (Default: false)",
```
Do we have to mention that?

hickeyma replied:
Same comment as above flag

hickeyma commented:
Thanks for review and comments @saschagrunert and @mikebrow. Updated and ready for review again.

hickeyma commented:
Thanks for the review comments @mikebrow. I have updated the debug message as suggested.

> also in testing there's an odd issue here where the config timeout format must be int and the --timeout option must be int+s for example 2s not 2.

This is a tricky one because if I change this so that both the CLI flag and the config file property use the same data type then I would break backwards compatibility. I added some extra text to flag and doc to be more explicit on it. Hope this helps.

hickeyma commented:
Thanks for review @mikebrow @saschagrunert @feiskyer and for merging @feiskyer
